### PR TITLE
Register the recipient for master-master sharing

### PR DESCRIPTION
**Pull request by Gara64**
In a Master-Master sharing context, the recipient must register an OAuth client to the sharer, as it will be necessary for the recipient to have access on the sharer docs.
This PR implements this registration, just after the sharing request is received on the recipient side. It then sends the generated `client_id` to the sharer, for him to save it in his sharing document.

Note this relies on #502

### CODE REVIEW

File: pkg/sharings/send_mails.go [hunk 1]
```diff
@@ -43,54 +43,58 @@ func SendSharingMails(instance *instance.Instance, s *Sharing) error {
...
-		// Sanity check: recipient is private.
-		if rs.recipient == nil {
-			rs.recipient, err = GetRecipient(instance, rs.RefRecipient.ID)
-			if err != nil {
-				return err
+
+		// A non-empty status indicates a recipient already treated
+		if rs.Status != "" {
+			// Sanity check: recipient is private.
+			if rs.recipient == nil {
```
Comment (aenario) [inline pkg/sharings/send_mails.go]: this code is repeated, might be worth it to make it a method on rs.

File: pkg/sharings/sharings.go [hunk 2]
```diff
@@ -408,10 +410,77 @@ func CreateSharingRequest(db couchdb.Database, desc, state, sharingType, scope,
...
 }
 
-// CreateSharingAndRegisterSharer checks the sharing, creates the document in
+// RegisterRecipient registers a sharing recipient
+func RegisterRecipient(instance *instance.Instance, rs *RecipientStatus) error {
+	err := rs.Register(instance)
+	if err != nil {
+		if rs.recipient != nil {
+			log.Error("[sharing] Could not register at "+
+				rs.recipient.URL+" ", err)
```
Comment (jinroh) [inline pkg/sharings/sharings.go]: nit: FYI, you can use `log.Errorf` for this kind of format.

File: pkg/sharings/send_mails.go (hunk 1 above)
Comment (Gara64) [inline pkg/sharings/send_mails.go]: Yup, I just added the method

File: pkg/sharings/sharings.go (hunk 2 above)
Comment (Gara64) [inline pkg/sharings/sharings.go]: Okay, thanks for the tip

**Pull request merged**